src/python/WMCore/Services/Rucio/Rucio.py

```diff
@@ -229,3 +239,246 @@ def getPFN(self, nodes=None, lfns=None, destination=None, protocol='srmv2', cust
+        return response
+
+    def createReplicationRule(self, names, rseExpression, scope='cms', copies=1, **kwargs):
+        """
+        _createReplicationRule_
+
+        Creates a replication rule against a list of data identifiers.
+        :param names: either a string with a did or a list of dids of the same type
+          (either a block or a container name)
+        :param rseExpression: boolean string expression to give the list of RSEs.
+           Full documentation on: https://rucio.readthedocs.io/en/latest/rse_expressions.html
+           E.g.: "tier=2&US=true"
+        :param scope: string with the scope name
+        :param kwargs:  supported keyword arguments (from the Rucio CLI API documentation):
+           * weight:    If the weighting option of the replication rule is used,
+             the choice of RSEs takes their weight into account.
+           * lifetime:  The lifetime of the replication rules (in seconds).
+           * grouping:  Decides how the replication is going to happen; where:
+                ALL: All files will be replicated to the same RSE.
+                DATASET: All files in the same block will be replicated to the same RSE.
+                NONE: Files will be completely spread over all allowed RSEs without any
+                grouping considerations at all.
+                E.g.: ALL grouping for 3 blocks against 3 RSEs. All 3 blocks go to one of the RSEs
+                      DATASET grouping for 3 blocks against 3 RSEs, each block gets fully replicated
+                      to one of those random RSEs
+           * account:   The account owning the rule.
+           * locked:    If the rule is locked, it cannot be deleted.
+           * source_replica_expression: RSE Expression for RSEs to be considered for source replicas.
+           * activity:  Transfer Activity to be passed to FTS.
+           * notify:    Notification setting for the rule (Y, N, C).
+           * purge_replicas: When the rule gets deleted purge the associated replicas immediately.
+           * ignore_availability: Option to ignore the availability of RSEs.
+           * ask_approval: Ask for approval of this replication rule.
+           * asynchronous: Create rule asynchronously by judge-injector.
+           * priority:  Priority of the transfers.
+           * comment:   Comment about the rule.
+           * meta:      Metadata, as dictionary.
+        :return: a boolean to represent whether it succeeded or not
+        """
+        kwargs.setdefault('grouping', 'ALL')
+        kwargs.setdefault('account', self.rucioParams.get('account'))
+        kwargs.setdefault('locked', False)
+        kwargs.setdefault('notify', 'N')
+        kwargs.setdefault('purge_replicas', False)
+        kwargs.setdefault('ignore_availability', False)
+        kwargs.setdefault('ask_approval', False)
+        kwargs.setdefault('asynchronous', False)
+        kwargs.setdefault('priority', 3)
+
+        if isinstance(names, basestring):
```

Comment:
This is not going to work in python3. Is there another way?

Reply:
I will check if it's not a list instead ;)